src/libraries/System.Security.Cryptography/tests/MLDsaOpenSslTests.Unix.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Security.Cryptography.Tests
+{
+    [ConditionalClass(typeof(MLDsa), nameof(MLDsa.IsSupported))]
+    public sealed class MLDsaOpenSslTests : MLDsaTestsBase
+    {
+        protected override MLDsa GenerateKey(MLDsaAlgorithm algorithm)
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.MLDsaGenerateKey(algorithm.Name, ReadOnlySpan<byte>.Empty);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportPrivateSeed(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.MLDsaGenerateKey(algorithm.Name, source);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.EvpPKeyFromData(algorithm.Name, source, privateKey: true);
+            return new MLDsaOpenSsl(key);
+        }
+
+        protected override MLDsa ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source)
+        {
+            using SafeEvpPKeyHandle key = Interop.Crypto.EvpPKeyFromData(algorithm.Name, source, privateKey: false);
+            return new MLDsaOpenSsl(key);
+        }
+
+        [Fact]
+        public void MLDsaOpenSsl_Ctor_ArgValidation()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("pkeyHandle", static () => new MLDsaOpenSsl(null));
+        }
+
```

Comment:
```C#

        [Fact]
        public void MLDsaOpenSsl_Ctor_InvalidHandle()
        {
            AssertExtensions.ThrowsAny<CryptographicException>(static () => new MLDsaOpenSsl(new SafeEvpPKeyHandle()));
        }
```

Should be reliable.

Reply:
I went with ECDsaOpenSsl exception type (ArgumentException)